### PR TITLE
Support rename db for collection

### PR DESCRIPTION
**Pull request by jaime0815**
issue #25806

### CODE REVIEW

File: internal/rootcoord/meta_table.go [hunk 1]
```diff
@@ -705,22 +706,35 @@ func (mt *MetaTable) RenameCollection(ctx context.Context, dbName string, oldNam
...
-	_, ok := mt.aliases.get(dbName, oldName)
+	_, ok = mt.aliases.get(dbName, oldName)
 	if ok {
 		log.Warn("unsupported use a alias to rename collection")
 		return fmt.Errorf("unsupported use an alias to rename collection, alias:%s", oldName)
 	}
 
 	// check new collection already exists
-	newColl, err := mt.getCollectionByNameInternal(ctx, dbName, newName, ts)
+	newColl, err := mt.getCollectionByNameInternal(ctx, newDBName, newName, ts)
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:737]: If the user rename a collection without changing the db, this check will fail.

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: internal/rootcoord/meta_table.go (hunk 1 above)
Comment (longjiquan) [reply internal/rootcoord/meta_table.go:737]: Ignore this, in fact I just thinked about the case "rename a collection without changing the db name and collection name".
However, by the way, how do we treat this case? I didn't notice the logic handing this case here.

Comment (jaime0815) [reply internal/rootcoord/meta_table.go:737]: Yes, no matter the collection whether changing the db name, the target collection name should not be in target db

Comment (longjiquan) [reply internal/rootcoord/meta_table.go:737]: So, what about my second question?
It seems strange that we report this error, "duplicated new collection name :%s with other collection name or alias".

**Pull request merged**